bundles/com.espressif.idf.debug.gdbjtag.openocd/src/com/espressif/idf/debug/gdbjtag/openocd/ui/AppLvlTracingHandler.java

```diff
@@ -0,0 +1,149 @@
+package com.espressif.idf.debug.gdbjtag.openocd.ui;
+
+import org.eclipse.cdt.debug.core.ICDTLaunchConfigurationConstants;
+import org.eclipse.core.commands.AbstractHandler;
+import org.eclipse.core.commands.ExecutionEvent;
+import org.eclipse.core.commands.ExecutionException;
+import org.eclipse.core.expressions.IEvaluationContext;
+import org.eclipse.core.resources.IResource;
+import org.eclipse.core.runtime.CoreException;
+import org.eclipse.core.runtime.IProgressMonitor;
+import org.eclipse.core.runtime.IStatus;
+import org.eclipse.core.runtime.Status;
+import org.eclipse.core.runtime.jobs.Job;
+import org.eclipse.debug.core.DebugPlugin;
+import org.eclipse.debug.core.ILaunchConfiguration;
+import org.eclipse.debug.core.ILaunchManager;
+import org.eclipse.debug.internal.core.LaunchConfiguration;
+import org.eclipse.jface.dialogs.MessageDialog;
+import org.eclipse.jface.wizard.WizardDialog;
+import org.eclipse.launchbar.ui.NewLaunchConfigWizard;
+import org.eclipse.launchbar.ui.NewLaunchConfigWizardDialog;
+import org.eclipse.swt.widgets.Display;
+import org.eclipse.swt.widgets.Shell;
+import org.eclipse.ui.IPageLayout;
+import org.eclipse.ui.PlatformUI;
+
+import com.espressif.idf.core.logging.Logger;
+import com.espressif.idf.debug.gdbjtag.openocd.dsf.LaunchConfigurationDelegate;
+import com.espressif.idf.ui.handlers.EclipseHandler;
+import com.espressif.idf.ui.tracing.AppLvlTracingDialog;
+
+import ilg.gnumcueclipse.debug.gdbjtag.DebugUtils;
+
+@SuppressWarnings("restriction")
+public class AppLvlTracingHandler extends AbstractHandler
+{
+
+	private IResource project;
+
+	@Override
+	public Object execute(ExecutionEvent event) throws ExecutionException
+	{
+		Shell activeShell = PlatformUI.getWorkbench().getActiveWorkbenchWindow().getShell();
```

Comment:
Maybe we could use EclispeUtil.getShell()

Reply:
done